### DISCUSSION TITLE
PfSense/Netgate supported ?

### DISCUSSION THREAD

**Discussion by Astorias96** [CactuseSecurity/firewall-orchestrator#2289] [Q&A]
Hello there,

I just found this solution which looks very nice ! I tried to figure it out but could not find it in the documentation yet: Is PfSense/Netgate supported ? Is there any way to add custom integrations if it is not supported ?

Many thanks in advance for your feedback.
Kind regards,
Astorias96

**Comment by tpurschke** [comment]
Hi Astorias96,

currently pfSense / OPNSense ist not supported but an import module might be implemented.
At least for OPNsense the effort is not to great, since the  API is trivial and consists of a single call for getting the whole config.
Not sure about pfSense but there currently does not seem to be any API (documentation).
Of course you can write your own import module for pfSense and integrate it via a pull request. 
You simply would need to find a way to read out the firewall configuration remotely. SSH might be an option though this is deemed legacy support in Firewall Orchestrator.

See also #2135 

What is your use case? How many firewalls and rules do you want to integrate? Would switching to OPNSense be an option?
